src/sagemaker/workflow/_utils.py

```diff
@@ -57,7 +60,7 @@ def __init__(
         description: str = None,
         source_dir: str = None,
         dependencies: List = None,
-        depends_on: Union[List[str], List[Step]] = None,
+        depends_on: Optional[List[Union[str, Step, "StepCollection"]]] = None,
```

Comment:
This is to prevent the circular import
The type annotations in `Step` and `StepCollection` etc. can cause circular import errors since they're referencing each other.
The fix for this kind of error is to do a type checking import, which requires to add the "" quote on lower version of python
```
if TYPE_CHECKING:
    from sagemaker.workflow.steps import Step
```

I saw somewhere mentioned that for python version below 3.7, if using TYPE_CHECKING, we still have to quote the type i.e. StepCollection here

Reply:
ok got you, thanks for explaining